src/webapp/src/auth/userSlice.ts

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { AuthStatuses, Login } from './types';
+import { signInUser, signOutUser } from './asyncAuthActions';
+
+const initialState: Login = {
+  user: null,
+  error: null,
```

Comment:
we still dont know these values. `null` sometimes mean we know the value and is `null` whereas undefined means we still dont know what value is a variable having.

Reply:
I made them null because the return type from the `auth.signInWithPopup(provider)` function is `firebase.User | null` so the compiler was complaining about me using undefined. I found a solution. I think it looks better now. Let me know if you are not convinced. Whatever is cleaner I am open to use it! Thanks for your advice!